Time spine move from dbt-utils to dbt-core

matthewshaver opened the pull request:
Previews:
- [1.6](https://docs-getdbt-com-git-time-spine-dbt-labs.vercel.app/docs/build/metricflow-time-spine?version=1.6)
- [1.7](https://docs-getdbt-com-git-time-spine-dbt-labs.vercel.app/docs/build/metricflow-time-spine?version=1.7)

## What are you changing in this pull request and why?

Closes: https://github.com/dbt-labs/docs.getdbt.com/issues/4124

Move time_spine from dbt-utils to dbt-core


Uncomment if you're publishing docs for a prerelease version of dbt (delete if not applicable):
- [x] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [x] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

matthewshaver committed:
Time spine move from dbt-utils to dbt-core

matthewshaver commented:
@dbeatty10 can you tell me if this is an accurate way to represent this change?

dbeatty10 commented:
> @dbeatty10 can you tell me if this is an accurate way to represent this change?

Use `dbt` instead of `dbt_core`, and then you'll be good-to-go!

matthewshaver commented on website/docs/docs/build/metricflow-time-spine.md:
```diff
@@ -22,7 +22,7 @@ To create this table, you need to create a model in your dbt project called `met
 
     {{
-        dbt_utils.date_spine(
+        dbt_core.date_spine(
```
```suggestion
        dbt.date_spine(
```

matthewshaver commented on website/docs/docs/build/metricflow-time-spine.md:
```diff
@@ -45,7 +45,7 @@ select * from final
 {{config(materialized='table')}}
 with days as (
-    {{dbt_utils.date_spine(
+    {{dbt_core.date_spine(
```
```suggestion
    {{dbt.date_spine(
```

matthewshaver committed:
Apply suggestions from code review

matthewshaver committed:
Merge branch 'current' into time-spine

dbeatty10 reviewed (commented):
I think this will only work for dbt v1.7+

@Jstein77 or @QMalcolm do we need to preserve instructions the use `dbt_utils.date_spine` for dbt v1.6?

matthewshaver committed:
Merge branch 'current' into time-spine

QMalcolm commented:
> I think this will only work for dbt v1.7+
> 
> 
> 
> @Jstein77 or @QMalcolm do we need to preserve instructions the use `dbt_utils.date_spine` for dbt v1.6?

That is correct

matthewshaver committed:
Adding version blocks

matthewshaver committed:
Merge branch 'time-spine' of https://github.com/dbt-labs/docs.getdbt.com into time-spine

matthewshaver commented:
> > I think this will only work for dbt v1.7+
> > @Jstein77 or @QMalcolm do we need to preserve instructions the use `dbt_utils.date_spine` for dbt v1.6?
> 
> That is correct

Corrected this based on the feedback here. Added version blocks

matthewshaver committed:
Merge branch 'current' into time-spine

dbeatty10 commented on website/docs/docs/build/metricflow-time-spine.md:
```diff
@@ -38,8 +40,45 @@ final as (
+with days as (
+
+    {{
+        dbt.date_spine(
```
Remove duplicated line:
```suggestion
```

dbeatty10 committed:
Remove duplicated line

runleonarun reviewed (approved):
:shipit:

matthewshaver committed:
Merge branch 'current' into time-spine
